src/main/java/org/opensearch/ad/rest/handler/IndexAnomalyDetectorJobActionHandler.java

```diff
@@ -274,8 +339,9 @@ private void onIndexAnomalyDetectorJobResponse(IndexResponse response, AnomalyDe
      * 2.If job exists: a).if job state is disabled, return error message; b).if job state is enabled, disable job.
      *
      * @param detectorId detector identifier
+     * @param listener Listener to send responses
      */
-    public void stopAnomalyDetectorJob(String detectorId) {
+    public void stopAnomalyDetectorJob(String detectorId, ActionListener<AnomalyDetectorJobResponse> listener) {
```

Comment:
what is the purpose for having the listener on the stop detector job? currently the task state is already getting updated immediately after this action is finished. But with the new listener, wouldn't this trigger `indexAnomalyResult()` or `indexAnomalyResultException()`?